src/Php/ComposerPhpVersionFactory.php

```diff
@@ -0,0 +1,126 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Php;
+
+use Composer\Semver\VersionParser;
+use Nette\Utils\Json;
+use Nette\Utils\JsonException;
+use Nette\Utils\Strings;
+use PHPStan\File\CouldNotReadFileException;
+use PHPStan\File\FileReader;
+use PHPStan\ShouldNotHappenException;
+use function count;
+use function end;
+use function is_array;
+use function is_file;
+use function is_int;
+use function is_string;
+use function sprintf;
+
+class ComposerPhpVersionFactory
+{
+
+	private ?PhpVersion $minVersion = null;
+
+	private ?PhpVersion $maxVersion = null;
+
+	/**
+	 * @param string[] $composerAutoloaderProjectPaths
+	 * @param int|array{min: int, max: int}|null $phpVersion
+	 */
+	public function __construct(
+		private array $composerAutoloaderProjectPaths,
+		int|array|null $phpVersion,
+		bool $bleedingEdge,
+	)
+	{
+		if (is_int($phpVersion)) {
+			return;
```

Comment:
I think this if can be just removed.

Reply:
re-added this line, which fixed the build.